scripts/zkai_attention_d128_value_adapter_gate.py

```
@@ -0,0 +1,882 @@
+    _number(best.get("mean_abs_error"), "best mean abs error")
+    d128_input = _dict(analysis.get("d128_input"), "analysis d128 input")
+    target_pattern = _dict(d128_input.get("target_pattern"), "analysis target pattern")
+    if _bool(target_pattern.get("matches"), "target pattern match") is not True:
+        raise AttentionD128ValueAdapterError("target pattern evidence drift")
+    if expected is not None and _comparable(data) != _comparable(expected):
+        raise AttentionD128ValueAdapterError("payload content drift")
+
+    if key_set == FINAL_KEYS:
+        cases = _list(data.get("cases"), "cases")
+        if data.get("mutation_inventory") != list(EXPECTED_MUTATIONS):
+            raise AttentionD128ValueAdapterError("mutation inventory drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)